### PR TITLE
Feature/gradle clean up

### LINKED ISSUE #166: Remove Repetitive Setups in Gradle Files
With the new build plugins approach, we already have most stuff set up eg target SDK, and don't need to set them again in the Gradle files. Look for all the things that can be cleaned up

### PR DESCRIPTION
**Pull request by janewaitara**
# Scope

_Please make sure to read the [Contribution Guidelines](https://github.com/droidconKE/droidconKE2023Android/blob/main/CONTRIBUTING.md)
and check that you understand and have followed it as best as possible Explain what your feature
does in a short paragraph._ please check the below boxes
- [x] I have followed the coding conventions
- [ ] I have added/updated necessary tests
- [x] I have tested the changes added on a physical device
- [x] I have run `./codeAnalysis.sh` on linux/unix or `codeAnalysys.bat` on windows to make sure all lint/formatting checks have been done.

## Closes/Fixes Issues
 Closes #166

### CODE REVIEW

**Review by kibettheophilus — APPROVED** [review]

File: app/build.gradle.kts [hunk 1]
```diff
@@ -70,11 +67,8 @@ dependencies {
     implementation(libs.timber)
 
     implementation(libs.work.runtime)
-    implementation(libs.hilt.work)
-    implementation(libs.hilt.common)
 
     androidTestImplementation(libs.android.test.junit4)
```
Comment (chepsi) [inline app/build.gradle.kts:71]: do we need this?

File: data/build.gradle.kts [hunk 2]
```diff
@@ -50,20 +40,12 @@ dependencies {
...
     implementation(libs.datastore)
     implementation(libs.kotlin.coroutines.android)
-    implementation(libs.ktor.core)
-    implementation(libs.ktor.android)
-    implementation(libs.ktor.content.negotiation)
-    implementation(libs.ktor.json)
-    implementation(libs.ktor.auth)
-    implementation(libs.ktor.logging)
-    implementation(libs.ktor.okhttp)
+    implementation(libs.bundles.ktor)
```
Comment (chepsi) [inline data/build.gradle.kts:43]: 🏅

**Review by chepsi — APPROVED** [review]
Great job.

**Review by wangerekaharun — APPROVED** [review]

**Review by MamboBryan — APPROVED** [review]

File: app/build.gradle.kts (hunk 1 above)
Comment (janewaitara) [reply app/build.gradle.kts:71]: Yes we do.. I moved them to the hilt convention plugin.

File: data/build.gradle.kts (hunk 2 above)
Comment (janewaitara) [reply data/build.gradle.kts:43]: 💃

**Pull request merged**